File: microservice-course-management/src/main/java/com/vicentemartinez/microservicecoursemanagement/service/CourseService.java
package com.vicentemartinez.microservicecoursemanagement.service;

import java.util.List;

import com.vicentemartinez.microservicecoursemanagement.model.Course;
import com.vicentemartinez.microservicecoursemanagement.model.Transaction;

public interface CourseService {
	List<Course> allCourses();

	List<Course> filterCoursesByIdList(List<Long> idList);

	List<Course> filterCourses(String content);

	List<Transaction> filterTransactionsOfUser(Long userId);

	List<Transaction> filterTransactionsOfCourse(Long courseId);

	void saveTransaction(Transaction transaction);

	Course findCourseById(Long courseId);
}

File: course-enrollment-client/src/app/services/course.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Course } from "../model/Course";
import { Transaction } from "./../model/Transaction";
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';

let API_URL = 'http://localhost:8765/api/course/service/';

@Injectable({
  providedIn: 'root'
})
export class CourseService {

  constructor(private http: HttpClient) { }

  enroll(transaction: Transaction): Observable<any> {
    return this.http.post(API_URL + "enroll", JSON.stringify(transaction),
      {
        headers: {
          "Content-Type": "application/json; charset-UTF-8"
        }
      }
    );
  }

  findAllCourses(): Observable<any> {
    return this.http.post(API_URL,
      {
        headers: {
          "Content-Type": "application/json; charset-UTF-8"
        }
      }
    );
  }

  filterStudents(courseId: string): Observable<any> {
    return this.http.post(API_URL + "students", courseId,
      {
        headers: { "Content-Type": "application/json; charset-UTF-8" }
      }
    );
  }

  popularCourses(): Observable<any> {
    return this.http.get(API_URL + "popular",
      {
        headers: { "Content-Type": "application/json; charset=UTF-8" }
      }
    );
  }

  filterCourses(text: string): Observable<any> {
    return this.http.post(API_URL + "filter", text,
      {
        headers: { "Content-Type": "application/json; charset=UTF-8" }
      }
    );
  }
}
